src/macos/x86_64/vcpu.rs

```diff
@@ -624,8 +630,8 @@ impl XhyveCpu {
 		self.vcpu.write_register(&Register::R13, 0)?;
 		self.vcpu.write_register(&Register::R14, 0)?;
 		self.vcpu.write_register(&Register::R15, 0)?;
-		self.setup_system_gdt()?;
-		self.setup_system_64bit()?;
+		self.setup_system_gdt(guest_address)?;
+		self.setup_system_64bit(guest_address)?;
```

Comment:
These two functions directly use the offsets, so I added `guest_address` as a parameter. Still zero, just preparatory work.